exploratory_data_analysis/pubtator_exporatory_data_analysis.py

```diff
@@ -0,0 +1,257 @@
+                doi = doc_obj.xpath("passage/infon[@key='article-id_doi']/text()")
+                pmc_id = doc_obj.xpath("passage/infon[@key='article-id_pmc']/text()")
+                pmid = doc_obj.xpath("passage/infon[@key='article-id_pmid']/text()")
+                sections = doc_obj.xpath("passage/infon[@key='section_type']/text()")
+                section_type = doc_obj.xpath("passage/infon[@key='type']/text()")
+                year = doc_obj.xpath("passage/infon[@key='year']/text()")
+
+                section_headers = "|".join(sorted(list(set(sections))))
+                section_headers_type = "|".join(
+                    sorted(list(set(map(lambda x: x.upper(), section_type))))
+                )
+
+                writer.writerow(
+                    {
+                        "batch_folder": batch_directory.name,
+                        "doc_id": doc_id[0],
+                        "doi": doi[0] if len(doi) > 0 else "",
+                        "pmc": pmc_id[0] if len(pmc_id) > 0 else "",
+                        "pmid": pmid[0] if len(pmid) > 0 else "",
+                        "section": section_headers
+                        if len(section_headers) > 0
+                        else section_headers_type,
+                        "published_year": year[0]
+                        if len(year) > 0
+                        else 0,  # Shouldnt get to the else statement
+                    }
+                )
+
+
+# # Analyze Abstract/Full Text Dataset
+
+pubtator_central_metadata_df = pd.read_csv("output/pmc_metadata.tsv.xz", sep="\t")
+print(pubtator_central_metadata_df.shape)
+pubtator_central_metadata_df.head()
+
+# ## Sanity Check the data
+
+# Sanity check that all documents have a published year
+(pubtator_central_metadata_df.query("published_year.isnull()").shape)
+
+# Do all documents have an id?
+(pubtator_central_metadata_df.query("doc_id.isnull()").shape)
+
+# Do all documents have a pmid?
+(pubtator_central_metadata_df.query("pmid.isnull()").shape)
+
+# Do all documents have a pmc id?
+(pubtator_central_metadata_df.query("pmc.isnull()").shape)
+
+# Do all documents have a doi?
+(pubtator_central_metadata_df.query("doi.isnull()").shape)
+
+# ## Published Year Distribution
+
+(pubtator_central_metadata_df.sort_values("published_year").published_year.unique())
+
+doc_count_df = (
+    pubtator_central_metadata_df.groupby("published_year")
+    .agg({"published_year": "size"})
+    .rename(index=str, columns={"published_year": "doc_count"})
+    .reset_index()
+    .astype({"published_year": int, "doc_count": int})
+)
+doc_count_df.head()
+
+g = (
+    p9.ggplot(
+        doc_count_df.query("published_year > 0& published_year < 1950"),
+        p9.aes(x="published_year", y="doc_count"),
+    )
+    + p9.geom_col(position=p9.position_dodge(width=0.9), fill="#1f78b4")
+    + p9.labs(
+        title="Number of Abstracts Pre 1950", x="Publication Year", y="Document Count"
```

Comment:
Is there a reason you chose 1950? Looks like thats when publications started to increase, any reason you can think of? I'm just curious..

Reply:
>Is there a reason you chose 1950? Looks like thats when publications started to increase, any reason you can think of? I'm just curious..

Oh man that is a great question. Unfortunately, I don't have a great answer for you as  I currently don't have enough background knowledge. I just picked 1950 as a cutoff, but it would be awesome if I can discover that answer later in this project.